### PR TITLE
Add r-mlpack recipe

### PR DESCRIPTION
**Pull request by coatless**
cc @conda-forge/help-r

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Related to mlpack/mlpack#2738

Created using: https://github.com/bgruening/conda_r_skeleton_helper

### CODE REVIEW

**Comment by coatless** [comment]
On win_64, I'm seeing:

```bash
ValueError: Incompatible component merge:
  - '11_h8933c1f_netlib'
  - '21_mkl'
```

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=254268&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=23010c01-6a9d-5f94-9037-792618644b77&l=69

**Comment by coatless** [comment]
Should be ready for review!

File: recipes/r-mlpack/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    - r-rcpparmadillo >=0.8.400.0
+    - r-rcppensmallen >=0.2.10.0
+    - libblas
+    - libgomp      # [linux]
+    - liblapack
+
+  run:
+    - r-base
+    - {{ native }}gcc-libs         # [win]
+    - r-bh >=1.58
```
Comment (dbast) [inline recipes/r-mlpack/meta.yaml:50]: Not required as header only.
```suggestion
    #- r-bh >=1.58
```

**Comment by coatless** [comment]
@dbast why is a header-only include package not required? (e.g. `r-bh` was dropped?)

**Comment by dbast** [comment]
It is only required during compiling the package... but not to run (load) the built package...

**Pull request merged**